### PR TITLE
Image resizer

### PR DESCRIPTION
**Pull request by emord**
http://manage.dimagi.com/default.asp?181411#1010439

Still needs to be version flagged, but this is most of it

@orangejenny @kaapstorm

### CODE REVIEW

**Comment by emord** [comment]
to be merged with https://github.com/dimagi/commcare-hq/pull/8224

File: src/mugs.js [hunk 1]
```diff
@@ -1508,6 +1546,7 @@ define([
             // Nestable Field List not supported in CommCare before v2.16
             group_in_field_list = opts.features.group_in_field_list,
             richText = opts.features.rich_text;
+        ENABLE_IMAGE_RESIZER = opts.features.image_resize;
```
Comment (orangejenny) [inline src/mugs.js]: Is it feasible to set a `resizeEnabled` flag on the Image mug instead of using a global flag?

File: src/widgets.js [hunk 2]
```diff
@@ -653,6 +657,14 @@ define([
             });
         };
 
+        if (options.defaultOptions) {
+            widget.addOptions(options.defaultOptions);
+
+            if (options.defaultValue) {
```
Comment (orangejenny) [inline src/widgets.js]: Not quite following why this was added...should the spec in mugs.js include a defaultValue?

**Comment by orangejenny** [comment]
looks good, a couple of minor questions

File: src/widgets.js (hunk 2 above)
Comment (emord) [inline src/widgets.js]: That was the original way I implemented it, but forgot to clean it up

File: src/mugs.js (hunk 1 above)
Comment (emord) [inline src/mugs.js]: Done

**Pull request merged**